test/Microsoft.AspNet.Mvc.ViewFeatures.Test/Rendering/HtmlHelperTextBoxTest.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq.Expressions;
+using Microsoft.AspNet.Mvc.ModelBinding;
+using Microsoft.AspNet.Mvc.TestCommon;
+using Microsoft.AspNet.Mvc.ViewFeatures;
+using Microsoft.AspNet.Testing;
+using Xunit;
+using Microsoft.AspNet.Mvc.ModelBinding.Metadata;
+
+namespace Microsoft.AspNet.Mvc.Rendering
+{
+    public class HtmlHelperTextBoxTest
```

Comment:
Nice test! Would be good to also add negative tests: That is, make sure that even with a `Prompt` set, that "incompatible" input types (ones not in the list you created) do *not* get placeholder text.